src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -87,9 +87,14 @@ pub trait Encoder {
     fn delta_base(block: &IndexBlock) -> t_docId {
         block.last_doc_id
     }
+}
+
+/// Trait to model that an encoder can be decoded by a decoder.
+pub trait DecodedBy: Encoder {
+    type Decoder: Decoder;
 
     /// Create a new decoder that can be used to decode records encoded by this encoder.
-    fn decoder() -> impl Decoder;
+    fn decoder() -> Self::Decoder;
 }
```

Comment:
What's the benefit of splitting this out?

Reply:
I explained it in the commit message.

Reply:
I see, thanks!!